### PR TITLE
refactor: Pass ArgsManager into functions that register args

### LINKED ISSUE #19511: Pass ArgsManager into functions that register args with AddArg
Bitcoin Core has a single gArgs, which is used by all binaries (bitcoind, bitcoin-qt, bench_bitcoin, bitcoin-wallet, ...). This is generally fine, because binaries only need to parse arguments for themselves.

However, for testing purposes in bench_bitcoin, some benchmarks (at the time of writing I believe two) spin up a Bitcoin Core node via the TestingSetup NodeContext. Thus, they will end up with conflicting use of gArgs. This issue is fixed in https://github.com/bitcoin/bitcoin/pull/18662

Moreover, test_bitcoin might want to set up and pass around local `ArgsManager`s without modifying the "main" global gArgs. Also, some args are parsed from gArgs inside the mempool or other validation code, which also makes it hard for unit tests to mock those settings. Slowly getting rid of usages of gArgs is going to help with that.

And finally, globals usually come with other issues such as the static initialization fiasco. Getting rid of them will make the code cleaner.

I started doing that in commit fa6b97ce38baaa91991d8efa3e32d260fea298fa, which can serve as an example of how to do it for the other functions.

They can be found with `git grep -W 'gArgs.AddArg("'`.

#### Useful skills:

Basic knowledge of how to compile Bitcoin Core, modify its source code and contribute patches via git on GitHub.

#### Want to work on this issue?

For guidance on contributing, please read [CONTRIBUTING.md](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md) before opening your pull request.

### PR DESCRIPTION
**Pull request by S3RK**
Rationale: reduce use of gArgs to decouple code and simplify future maintenance and easier unit testing.

This PR is continuation of work started in  #18926 and #18662
It covers only places that register args in ArgsManager with `AddArgs()` or `AddHiddenArgs()`.

Closes #19511

### CODE REVIEW

File: src/chainparams.cpp [hunk 1]
```diff
@@ -341,8 +341,8 @@ class CRegTestParams : public CChainParams {
 
 void CRegTestParams::UpdateActivationParametersFromArgs(const ArgsManager& args)
 {
-    if (gArgs.IsArgSet("-segwitheight")) {
-        int64_t height = gArgs.GetArg("-segwitheight", consensus.SegwitHeight);
+    if (args.IsArgSet("-segwitheight")) {
```
Comment (S3RK) [inline src/chainparams.cpp:344]: This line had been added when local `args` already existed. I couldn't find any reason for the use of global variable, but maybe I'm missing something.

**Comment by MarcoFalke** [comment]
scripted diffs shouldn't modify out-of-git-tree files. Please replace the wildcard; See other scripted diffs for examples.

**Comment by MarcoFalke** [comment]
Concept ACK, apart from my feedback ofc

**Comment by S3RK** [comment]
Updated scripted diff to avoid touching out-of-tree files

**Comment by MarcoFalke** [comment]
ACK 8ed9002cd14165f751442f738fbf1fb8a37611b2 👛

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

ACK 8ed9002cd14165f751442f738fbf1fb8a37611b2 👛
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUh2ZAwAnz4O1skNtfv/RUWHkHJzWfJq4jaAvEdKEd3Mgd8q2rAU8bvUGlrEwqCa
8ljMbgcBZbQVndx9E5QomFlEHx+C3TDjxDpU66sh5TM2qIQ7gGDA6NmMb5pabSRx
bNG1OGDclMN2PiWWmtvKvKx5XgGUb3O7h1gpl3NJJsC/52RvU/mcY3Is9VoLwEs3
9rD9RM5pE7VLa9l2tZiaQrp7d1EHeqKtgcIAYCoChACi44wV+HYdtQa0kzwg2LhO
PMhtjws0damzM899I8PGApDjLlDTm61ShQJkoO42LxIHt4Sjs38kBl209MqixEd2
Vhsw7cDFcCmqALLOXP6zuhd1cq3NY4q31e2cL3RjhnxOz1gTmirQrWZXH8Y8MmdC
/2XT3E0kFKp+FGu5uL8WjS8/5bq0mfSGPfp7WCVEsFdRZdOdfNlrqdiTpSa9apXO
DHEzLehro9PCsVT64GelPwGaL9Y6NKEUqu5nYo0WnLTwUzpOvOePYfuAoLFeOVsy
x0xmmPVv
=CHO8
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `e25142d47fe5c4ec133bc1f892c23685dd05953099c70d44b4fe8f4bd41adb03  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e892940108e25142d47fe5c4ec133bc1f892c23685dd05953099c70d44b4fe8f4bd41adb03f0108ccba5bd738c03177e07fac91495a32208fff01004c2b40860b640d2b6d2670f2e21e4f108f1045f22983ef00889573943665448870083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010b6bdd819c4f59bec615cff6b5b94410c08f1045f22983cf0085df64764d6ea696d0083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff010cbc51bd392935f35c6a375b31a48873208f1045f22983ff0087d44927e8e9d56ed0083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010f9829049ea91f17388807df2f3e1c51c08f1045f22983bf008283677dc5b1ae1090083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6d)

</details>

**Pull request merged**